### PR TITLE
[Merged by Bors] - feat: homeomorphism between `C(X, Σ i, Y i)` and `Σ i, C(X, Y i)`

### PR DESCRIPTION
**Pull request by urkud**
This is a follow-up to #4325.

---

- [x] depends on: #5649 
- [x] depends on: #5652

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#5649
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#5649~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#5649~~
* ~~leanprover-community/mathlib4#5652~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Topology/Constructions.lean [hunk 1]
```diff
@@ -1555,6 +1555,22 @@ theorem isOpenMap_sigma_map {f₁ : ι → κ} {f₂ : ∀ i, σ i → τ (f₁
     forall_congr' fun i => (@openEmbedding_sigmaMk _ _ _ (f₁ i)).isOpenMap_iff.symm
 #align is_open_map_sigma_map isOpenMap_sigma_map
 
+/-- A map defined on a sigma type (a.k.a. the disjoint union of an indexed family of topological
+spaces) is inducing iff its restriction to each component is inducing and each the image of each
+component under `f` can be separated from the images of all other components by an open set. -/
+theorem inducing_sigma {f : Sigma σ → α} :
+    Inducing f ↔ (∀ i, Inducing (f ∘ Sigma.mk i)) ∧
+      (∀ i, ∃ U, IsOpen U ∧ ∀ x, f x ∈ U ↔ x.1 = i) := by
```
Comment (ADedecker) [inline Mathlib/Topology/Constructions.lean:1563]: Could you move this closer to `continuous_sigma`? It feels a bit weird to have it in the middle of the results about `Sigma.map`

**Review by ADedecker — APPROVED** [review]
Otherwise LGTM!

**Comment by ADedecker** [comment]
Thanks!
maintainer merge

**Comment by ocfnash** [comment]
Thanks!

bors merge

**Pull request closed without merging**